pkg/scheduler/framework/plugins/dynamicresources/dynamicresources_test.go

```diff
@@ -0,0 +1,762 @@
+						WithAllocation(resourcev1alpha1ac.AllocationResult().
+							WithAvailableOnNodes(corev1ac.NodeSelector().
+								WithNodeSelectorTerms(corev1ac.NodeSelectorTerm().
+									WithMatchExpressions(corev1ac.NodeSelectorRequirement().
+										WithKey("no-such-label").
+										WithOperator(v1.NodeSelectorOpIn).
+										WithValues("no-such-value")))))))
+	allocatedClaimWithGoodTopology = toResourceClaim(resourcev1alpha1ac.ResourceClaim(claimName, namespace).
+					WithOwnerReferences(metav1ac.OwnerReference().
+						WithUID(podUID).
+						WithController(true)).
+					WithStatus(resourcev1alpha1ac.ResourceClaimStatus().
+						WithAllocation(resourcev1alpha1ac.AllocationResult().
+							WithAvailableOnNodes(corev1ac.NodeSelector().
+								WithNodeSelectorTerms(corev1ac.NodeSelectorTerm().
+									WithMatchExpressions(corev1ac.NodeSelectorRequirement().
+										WithKey("nodename").
+										WithOperator(v1.NodeSelectorOpIn).
+										WithValues("worker")))))))
+	otherClaim = toResourceClaim(resourcev1alpha1ac.ResourceClaim("not-my-claim", namespace))
+
+	schedulingSelectedPotential = toPodScheduling(resourcev1alpha1ac.PodScheduling(podName, namespace).
+					WithOwnerReferences(metav1ac.OwnerReference().
+						WithAPIVersion("v1").
+						WithController(true).
+						WithKind("Pod").
+						WithName(podName).
+						WithUID(podUID)).
+					WithSpec(resourcev1alpha1ac.PodSchedulingSpec().
+						WithSelectedNode(workerNode.Name).
+						WithPotentialNodes(workerNode.Name)))
+	schedulingPotential = toPodScheduling(resourcev1alpha1ac.PodScheduling(podName, namespace).
+				WithOwnerReferences(metav1ac.OwnerReference().
+					WithAPIVersion("v1").
+					WithController(true).
+					WithKind("Pod").
+					WithName(podName).
+					WithUID(podUID)).
+				WithSpec(resourcev1alpha1ac.PodSchedulingSpec().
+					WithPotentialNodes(workerNode.Name)))
+	schedulingInfo = toPodScheduling(resourcev1alpha1ac.PodScheduling(podName, namespace).
+			WithOwnerReferences(metav1ac.OwnerReference().
+				WithAPIVersion("v1").
+				WithController(true).
+				WithKind("Pod").
+				WithName(podName).
+				WithUID(podUID)).
+			WithSpec(resourcev1alpha1ac.PodSchedulingSpec().
+				WithPotentialNodes(workerNode.Name)).
+			WithStatus(resourcev1alpha1ac.PodSchedulingStatus().
+				WithResourceClaims(resourcev1alpha1ac.ResourceClaimSchedulingStatus().
+					WithName(resourceName), resourcev1alpha1ac.ResourceClaimSchedulingStatus().
+					WithName(resourceName2))))
+)
+
+// result defines the expected outcome of some operation. It covers return
+// values, the plugin state and the state of the world (= objects).
+type result struct {
+	returnValue interface{}
+	status      *framework.Status
+
+	// changes contains corev1ac apply configuration instances that contain
+	// the expected changes to existing objects. This way, new fields and
+	// can be described concisely. Removal of fields is not needed.
+	changes []any
```

Comment:
is there an interface that apply configurations implement?

Reply:
No, they don't, therefore some of the usual interfaces for working with Kubernetes objects don't work. It probably would make sense to add that. Added to https://github.com/kubernetes/kubernetes/issues/113773